src/client/src/tests/components/RegistrationForm.test.tsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { RegistrationForm } from '../../components/RegistrationForm';
+
+const renderForm = () => {
+    const createUser = jest.fn();
+    const registration = render(<RegistrationForm createUser={createUser} />);
+
+    return { createUser, registration };
+};
+
+describe('<RegistrationForm />', () => {
+    test('should be visible to users', () => {
+        const { registration } = renderForm();
+        const container = registration.container;
+        expect(container).toHaveTextContent('Register');
+
+        const inputs = container.querySelectorAll('input');
+        expect(inputs).toHaveLength(4);
+    });
+
+    test('should send a form given proper values', () => {
+        const { createUser, registration } = renderForm();
+
+        // Email, Username, Password, Confirm Password
+        const inputs = registration.container.querySelectorAll('input');
```

Comment:
These tests turn a bit neater if:
- Use an not-null assertion after these lines (so `const inputs = registration.container.querySelectorAll('input')!;`)
   - this way, you can skip the `if(form)` on line 49
- Use byLabelText & type:
   - https://testing-library.com/docs/ecosystem-user-event#typeelement-text-options
   -  userEvent.type(screen.getByLabelText('Enter Username'), 'username!')